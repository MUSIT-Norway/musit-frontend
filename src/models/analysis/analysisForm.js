// @flow
import type { Field } from '../../forms/form';
import type { AnalysisCollection } from '../../types/analysis';
import type { CollectionId } from 'types/ids';
import type { ImportAnalysisResult } from 'types/analysisResult';

export type FormValue = {
  name: string,
  defaultValue: ?string | boolean | Array<any>
};

const toField = (
  name: string,
  defaultValue: ?string | boolean | Array<any>
): FormValue => ({
  name,
  defaultValue
});

export const fromJsonToForm: (
  json: AnalysisCollection,
  fields: Array<Field<*>>
) => Array<FormValue> = (json, formDef) => {
  const formValues = formDef.reduce(
    (acc, field) => ({
      ...acc,
      [field.name]: toField(field.name, json[field.name])
    }),
    {}
  );

  let persons = [];
  if (formValues.doneBy && formValues.doneBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.doneByName,
        uuid: json.doneBy,
        role: 'doneBy',
        date: json.doneDate
      }
    ]);
  }

  if (formValues.responsible && formValues.responsible.defaultValue) {
    persons = persons.concat([
      {
        name: json.responsibleName,
        uuid: json.responsible,
        role: 'responsible',
        date: null
      }
    ]);
  }

  if (formValues.administrator && formValues.administrator.defaultValue) {
    persons = persons.concat([
      {
        name: json.administratorName,
        uuid: json.administrator,
        role: 'administrator',
        date: null
      }
    ]);
  }

  if (formValues.completedBy && formValues.completedBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.completedByName,
        uuid: json.completedBy,
        role: 'completedBy',
        date: json.completedDate
      }
    ]);
  }
  formValues.persons = toField('persons', persons);

  const restriction = json.restriction;
  formValues.restrictions = toField('restrictions', !!restriction);
  if (restriction) {
    formValues.restrictions_requester = toField(
      'restrictions_requester',
      restriction.requester
    );
    formValues.restrictions_requesterName = toField(
      'restrictions_requesterName',
      restriction.requesterName
    );
    formValues.restrictions_reason = toField('restrictions_reason', restriction.reason);
    formValues.restrictions_caseNumbers = toField(
      'restrictions_caseNumbers',
      restriction.caseNumbers
    );
    formValues.restrictions_cancelledReason = toField(
      'restrictions_cancelledReason',
      restriction.cancelledReason
    );
    formValues.restrictions_expirationDate = toField(
      'restrictions_expirationDate',
      restriction.expirationDate
    );
  }

  const result = json.result;
  if (result) {
    formValues.comments = toField('comments', result.comment);
    formValues.externalSource = toField('externalSource', result.extRef);
  }

  return Object.keys(formValues).map(key => formValues[key]);
};
