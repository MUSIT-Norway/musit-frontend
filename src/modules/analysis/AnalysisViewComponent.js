/* @flow */
import React from 'react';
import { I18n } from 'react-i18nify';
import { SaveCancel } from '../../components/formfields/index';
import { hashHistory } from 'react-router';
import Config from '../../config';
import type { AppSession } from '../../types/appSession';
import type { FormData } from './types/form';
import type { Store } from './types/store';
import { Table } from 'reactable';
import MetaInformation from './components/MetaInformation';

type Params = { analysisId: string };

type Props = { form: FormData, store: Store, appSession: AppSession, params: Params };

const AnalysisView = ({ form, store, appSession, params }: Props) => (
  <div>
    <div className="page-header">
      <h1>
        {I18n.t('musit.analysis.viewAnalysis')}
      </h1>
    </div>
    <form className="form-horizontal">
      <MetaInformation form={form} />
      <hr />
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="type">Type analyse:</label>
        <div className="col-md-10">
          <p className="form-control-static" id="type">
            {getAnalysisTypeTerm(form, store)}
          </p>
        </div>
      </div>
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="status">
          Status på analysen:
        </label>
        <div className="col-md-5">
          <p className="form-control-static" id="status">
            {getStatusText(form.status.value)}
          </p>
        </div>
      </div>
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="caseNumber">Saksnummer:</label>
        <div className="col-md-10">
          <p className="form-control-static" id="caseNumber">
            {form.caseNumber.value}
          </p>
        </div>
      </div>
      <hr />
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="note">
          Beskrivelse/ kommentar:
        </label>
        <div className="col-md-10">
          <p className="form-control-static" id="note">
            {form.note.value}
          </p>
        </div>
      </div>
      <div className="form-group">
        <label className="control-label">Personer tilknyttet analysen:</label>
      </div>
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="responsible-name1">
          Navn:
        </label>
        <div className="col-md-2">
          <p className="form-control-static" id="responsible-name1">n/a</p>
        </div>
        <label className="control-label col-md-1" htmlFor="responsible-role1">
          Rolle:
        </label>
        <div className="col-md-2">
          <p className="form-control-static" id="responsible-role1">n/a</p>
        </div>
      </div>
      <hr />
      <div className="well">
        <div className="form-group">
          <label className="control-label col-md-2" htmlFor="objects">
            Objekter:
          </label>
          <div className="col-md-10">
            <Table
              id="objects"
              className="table"
              columns={[
                { key: 'museumNo', label: 'Museumsnr' },
                { key: 'subNo', label: 'Unr' },
                { key: 'term', label: 'Term/artsnavn' }
              ]}
              data={
                form.type.value === 'AnalysisCollection'
                  ? form.events.value
                  : [
                      {
                        term: form.term.value,
                        museumNo: form.museumNo.value,
                        subNo: form.subNo.value
                      }
                    ]
              }
              sortable={['museumNumber', 'subNumber', 'term']}
              noDataText="Ingen objekter"
            />
          </div>
        </div>
        <hr />
        <div className="form-group">
          <label className="control-label col-md-2" htmlFor="externalSource">
            Ekstern kilde:
          </label>
          <div className="col-md-10">
            <p className="form-control-static" id="externalSource">
              {form.externalSource.value}
            </p>
          </div>
        </div>
        <div className="form-group">
          <label className="control-label col-md-2" htmlFor="externalFile">
            Vedlegg:
          </label>
          <div className="col-md-10">
            <p className="form-control-static" id="externalFile">{' '}</p>
          </div>
        </div>
        <div className="form-group">
          <label className="control-label col-md-2" htmlFor="comments">
            Kommentar til resultat:
          </label>
          <div className="col-md-10">
            <p className="form-control-static" id="comments">
              {form.comments.value}
            </p>
          </div>
        </div>
        <div className="form-group">
          <label className="control-label col-md-2" htmlFor="restrictions">
            Klausulering:
          </label>
          <div className="col-md-10">
            <p className="form-control-static" id="restrictions">
              {form.restrictions.value ? 'Ja' : 'Nei'}
            </p>
          </div>
        </div>
        {form.restrictions.rawValue &&
          <div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="requester">
                Klausulert for:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="requester">
                  {form.note.value || ''}
                </p>
              </div>
            </div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="reason">
                Årsak til klausulering:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="reason">
                  {form.reason.value || ''}
                </p>
              </div>
            </div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="caseNumbers">
                Saksnummer:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="caseNumbers">
                  {form.caseNumbers.value || ''}
                </p>
              </div>
            </div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="expirationDate">
                Sluttdato:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="expirationDate">
                  {form.expirationDate.value || ''}
                </p>
              </div>
            </div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="cancelledBy">
                Opphevet av:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="cancelledBy">
                  {form.cancelledBy.value || ''}
                </p>
              </div>
            </div>
            <div className="form-group">
              <label className="control-label col-md-2" htmlFor="cancelledReason">
                Årsak til oppheving:
              </label>
              <div className="col-md-10">
                <p className="form-control-static" id="cancelledReason">
                  {form.cancelledReason.value || ''}
                </p>
              </div>
            </div>
          </div>}
      </div>
      <hr />
      <SaveCancel
        onClickSave={e => {
          e.preventDefault();
          hashHistory.push(
            Config.magasin.urls.client.analysis.editAnalysis(
              appSession,
              params.analysisId
            )
          );
        }}
        saveLabel="Endre"
      />
    </form>
  </div>
);

function getStatusText(status?: ?number): string {
  if (!status) {
    return '';
  }
  switch (status) {
    case 1:
      return '1. Under forberedelse';
    case 2:
      return '2. Analyse påbegynt';
    case 3:
      return '3. Analyse ferdig';
    case 4:
      return '4. Avsluttet uten resultat';
    default:
      return 'N/A: ' + status;
  }
}

function getAnalysisTypeTerm(form, store) {
  if (form.analysisTypeId.rawValue && store.analysisTypes) {
    const foundType = store.analysisTypes.find(
      a => a.id === form.analysisTypeId.rawValue
    );
    return foundType ? foundType.name : '';
  }
}

export default AnalysisView;
