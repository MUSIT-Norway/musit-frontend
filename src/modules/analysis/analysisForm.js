// @flow
import createForm, {
  getStrField,
  getArrField,
  getBoolField,
  getNumberField
} from '../../forms/form';

export const fieldsArray = [
  getNumberField('id'),
  getStrField('analysisTypeId'),
  getStrField('doneBy'),
  getStrField('doneDate'),
  getStrField('registeredBy'),
  getStrField('registeredByName'),
  getStrField('registeredDate'),
  getStrField('responsible'),
  getStrField('responsible'),
  getStrField('administrator'),
  getStrField('completedBy'),
  getStrField('completedDate'),
  getStrField('objectId'),
  getStrField('note'),
  getStrField('type'),
  getStrField('partOf'),
  getStrField('result'),
  getStrField('place'),
  getStrField('reason'),
  getStrField('externalSource'),
  getStrField('comments'),
  getBoolField('restrictions', false),
  getStrField('restrictions_requester'),
  getStrField('restrictions_expirationDate'),
  getStrField('restrictions_reason'),
  getStrField('restrictions_cancelledReason'),
  getArrField('restrictions_caseNumbers'),
  getArrField('caseNumbers'),
  getBoolField('completeAnalysis'),
  getStrField('museumNo'),
  getStrField('subNo'),
  getStrField('term'),
  getArrField('events', []),
  getStrField('analysisTypeCategory'),
  getStrField('updatedBy'),
  getStrField('updatedByName'),
  getStrField('updatedDate'),
  getNumberField('status', 1, 1, 4, 0)
];

export default createForm('analysisForm.js', fieldsArray);
