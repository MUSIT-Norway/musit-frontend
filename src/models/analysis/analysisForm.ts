// @flow
import { Field } from '../../forms/form';
import { AnalysisCollection } from '../../types/analysis';
import { Maybe, Star, MUSTFIX, BUG } from '../../types/common';
import { Person } from '../../types/person';

export type FormValue = {
  name: string;
  defaultValue: Maybe<any>;
  rawValue?: Maybe<any>;
};

const toField = (name: string, defaultValue: Maybe<any>): FormValue => ({
  name,
  defaultValue
});

export const fromJsonToForm: (
  json: AnalysisCollection,
  fields: Array<Field<Star>>
) => Array<FormValue> = (json, formDef) => {
  const formValues: MUSTFIX = formDef.reduce(
    (acc, field) => ({
      ...acc,
      [field.name]: toField(field.name, json[field.name])
    }),
    {}
  );

  let persons = [] as Person[];
  if (formValues.doneBy && formValues.doneBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.doneByName as MUSTFIX,
        uuid: json.doneBy as MUSTFIX,
        role: 'doneBy',
        date: json.doneDate as MUSTFIX
      }
    ]);
  }

  if (formValues.responsible && formValues.responsible.defaultValue) {
    persons = persons.concat([
      {
        name: json.responsibleName as MUSTFIX,
        uuid: json.responsible as MUSTFIX,
        role: 'responsible',
        date: null as BUG
      }
    ]);
  }

  if (formValues.administrator && formValues.administrator.defaultValue) {
    persons = persons.concat([
      {
        name: json.administratorName as MUSTFIX,
        uuid: json.administrator as MUSTFIX,
        role: 'administrator',
        date: null as BUG
      }
    ]);
  }

  if (formValues.completedBy && formValues.completedBy.defaultValue) {
    persons = persons.concat([
      {
        name: json.completedByName as MUSTFIX,
        uuid: json.completedBy as MUSTFIX,
        role: 'completedBy',
        date: json.completedDate as MUSTFIX
      }
    ]);
  }
  formValues.persons = toField('persons', persons);

  const restriction = json.restriction;
  const hasRestriction = !!restriction && !restriction.cancelledStamp;
  formValues.restrictions = toField('restrictions', hasRestriction);
  formValues.restriction = toField('restriction', hasRestriction ? restriction : null);
  const result = json.result;
  if (result) {
    formValues.comments = toField('comments', result.comment);
    formValues.externalSource = toField('externalSource', result.extRef);
  }

  return Object.keys(formValues).map(key => formValues[key]);
};
