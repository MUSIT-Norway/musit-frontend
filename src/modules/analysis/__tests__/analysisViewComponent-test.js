// @flow
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import AnalysisViewComponent from '../AnalysisViewComponent';
import { fieldsArray } from '../analysisForm';
import type { Field } from 'forms/form';
import type { FormData } from '../types/form';

declare var describe: any;
declare var it: any;
declare var expect: any;

const objectsData = [
  {
    museumNumber: '123',
    subNumber: '12345678911',
    term: 'Spyd',
    uuid: '1cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '124',
    subNumber: '12345678912',
    term: 'Beltering',
    uuid: '2cbf15cb-8348-4e66-99a4-bc314da57a42'
  },
  {
    museumNumber: '125',
    subNumber: '12345678913',
    term: 'Ã˜sekar',
    uuid: '3cbf15cb-8348-4e66-99a4-bc314da57a42'
  }
];

const analysisTypes = [
  {
    category: 8,
    id: 1,
    name: '3D-skanning, laser'
  },
  {
    category: 8,
    id: 2,
    name: '3D-skanning, strukturert lys'
  }
];

const analysis = {
  analysisTypeId: '8453873d-227c-4205-a231-bf7e04164fab',
  eventDate: '2017-03-16T14:37:45+00:00',
  id: 2,
  museumNo: 'MusK58',
  note: 'fdsfsd sdsa 2',
  objectId: 'adea8141-8099-4f67-bff9-ea5090e18335',
  partOf: 1,
  registeredBy: '7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e',
  registeredByName: 'Rituvesh Kumar',
  registeredDate: '2017-04-03T10:36:34+00:00',
  subNo: '2',
  term: 'Mansjettknapp',
  type: 'Analysis'
};

const store = {
  analysisTypes: analysisTypes,
  objectsData: objectsData,
  analysis: analysis
};

const appSession = {
  museumId: 99,
  collectionId: '1234',
  accessToken: '45667',
  actor: {
    fn: 'Test'
  }
};

const form: FormData = (fieldsArray.reduce(
  (acc, field: Field<any>) => ({
    ...acc,
    [field.name]: {
      ...field,
      rawValue: field.mapper.toRaw(field.defaultValue)
    }
  }),
  {}
): any);

describe('AnalysisViewComponent', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <AnalysisViewComponent
        form={form}
        params={{ analysisId: '45' }}
        appSession={appSession}
        store={store}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
