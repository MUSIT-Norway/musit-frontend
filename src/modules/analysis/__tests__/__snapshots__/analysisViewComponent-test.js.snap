exports[`AnalysisViewComponent should render properly 1`] = `
<div>
  <div
    className="page-header">
    <h1>
      Analyse detaljer
    </h1>
  </div>
  <form
    className="form-horizontal">
    <MetaInformation
      form={
        Object {
          "administrator": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "administrator",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "analysisTypeCategory": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "analysisTypeCategory",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "analysisTypeId": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "analysisTypeId",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "cancelledBy": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "cancelledBy",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "cancelledReason": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "cancelledReason",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "caseNumber": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "caseNumber",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "caseNumbers": Object {
            "defaultValue": Array [],
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "caseNumbers",
            "rawValue": Array [],
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "comments": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "comments",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "completeAnalysis": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "completeAnalysis",
            "rawValue": false,
            "validator": Object {}
          },
          "completedBy": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "completedBy",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "completedDate": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "completedDate",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "doneBy": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "doneBy",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "doneDate": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "doneDate",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "events": Object {
            "defaultValue": Array [],
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "events",
            "rawValue": Array [],
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "expirationDate": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "expirationDate",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "externalSource": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "externalSource",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "id": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "id",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function configurableValidators],
              "valueValidator": [Function validator]
            }
          },
          "museumNo": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "museumNo",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "note": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "note",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "objectId": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "objectId",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "partOf": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "partOf",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "place": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "place",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "reason": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "reason",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "registeredBy": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "registeredBy",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "registeredByName": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "registeredByName",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "registeredDate": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "registeredDate",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "requester": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "requester",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "responsible": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "responsible",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "restrictions": Object {
            "defaultValue": false,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "restrictions",
            "rawValue": false,
            "validator": Object {}
          },
          "result": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "result",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "status": Object {
            "defaultValue": 1,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "status",
            "rawValue": "1",
            "validator": Object {
              "rawValidator": [Function configurableValidators],
              "valueValidator": [Function validator]
            }
          },
          "subNo": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "subNo",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "term": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "term",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "type": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "type",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "updatedBy": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "updatedBy",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "updatedByName": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "updatedByName",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          },
          "updatedDate": Object {
            "defaultValue": null,
            "mapper": Object {
              "fromRaw": [Function fromRaw],
              "toRaw": [Function toRaw]
            },
            "name": "updatedDate",
            "rawValue": null,
            "validator": Object {
              "rawValidator": [Function validator]
            }
          }
        }
      } />
    <hr />
    <div
      className="form-group">
      <label
        className="control-label col-md-2"
        htmlFor="type">
        Type analyse:
      </label>
      <div
        className="col-md-10">
        <p
          className="form-control-static"
          id="type" />
      </div>
    </div>
    <div
      className="form-group">
      <label
        className="control-label col-md-2"
        htmlFor="status">
        Status på analysen:
      </label>
      <div
        className="col-md-5">
        <p
          className="form-control-static"
          id="status" />
      </div>
    </div>
    <div
      className="form-group">
      <label
        className="control-label col-md-2"
        htmlFor="caseNumber">
        Saksnummer:
      </label>
      <div
        className="col-md-10">
        <p
          className="form-control-static"
          id="caseNumber" />
      </div>
    </div>
    <hr />
    <div
      className="form-group">
      <label
        className="control-label col-md-2"
        htmlFor="note">
        Beskrivelse/ kommentar:
      </label>
      <div
        className="col-md-10">
        <p
          className="form-control-static"
          id="note" />
      </div>
    </div>
    <div
      className="form-group">
      <label
        className="control-label">
        Personer tilknyttet analysen:
      </label>
    </div>
    <div
      className="form-group">
      <label
        className="control-label col-md-2"
        htmlFor="responsible-name1">
        Navn:
      </label>
      <div
        className="col-md-2">
        <p
          className="form-control-static"
          id="responsible-name1">
          n/a
        </p>
      </div>
      <label
        className="control-label col-md-1"
        htmlFor="responsible-role1">
        Rolle:
      </label>
      <div
        className="col-md-2">
        <p
          className="form-control-static"
          id="responsible-role1">
          n/a
        </p>
      </div>
    </div>
    <hr />
    <div
      className="well">
      <div
        className="form-group">
        <label
          className="control-label col-md-2"
          htmlFor="objects">
          Objekter:
        </label>
        <div
          className="col-md-10">
          <Table
            className="table"
            columns={
              Array [
                Object {
                  "key": "museumNo",
                  "label": "Museumsnr"
                },
                Object {
                  "key": "subNo",
                  "label": "Unr"
                },
                Object {
                  "key": "term",
                  "label": "Term/artsnavn"
                }
              ]
            }
            data={
              Array [
                Object {
                  "museumNo": undefined,
                  "subNo": undefined,
                  "term": undefined
                }
              ]
            }
            defaultSort={false}
            defaultSortDescending={false}
            filterBy=""
            hideFilterInput={false}
            id="objects"
            itemsPerPage={0}
            noDataText="Ingen objekter"
            sortBy={false}
            sortable={
              Array [
                "museumNumber",
                "subNumber",
                "term"
              ]
            } />
        </div>
      </div>
      <hr />
      <div
        className="form-group">
        <label
          className="control-label col-md-2"
          htmlFor="externalSource">
          Ekstern kilde:
        </label>
        <div
          className="col-md-10">
          <p
            className="form-control-static"
            id="externalSource" />
        </div>
      </div>
      <div
        className="form-group">
        <label
          className="control-label col-md-2"
          htmlFor="externalFile">
          Vedlegg:
        </label>
        <div
          className="col-md-10">
          <p
            className="form-control-static"
            id="externalFile">
             
          </p>
        </div>
      </div>
      <div
        className="form-group">
        <label
          className="control-label col-md-2"
          htmlFor="comments">
          Kommentar til resultat:
        </label>
        <div
          className="col-md-10">
          <p
            className="form-control-static"
            id="comments" />
        </div>
      </div>
      <div
        className="form-group">
        <label
          className="control-label col-md-2"
          htmlFor="restrictions">
          Klausulering:
        </label>
        <div
          className="col-md-10">
          <p
            className="form-control-static"
            id="restrictions">
            Nei
          </p>
        </div>
      </div>
    </div>
    <hr />
    <SaveCancel
      onClickSave={[Function onClickSave]}
      saveLabel="Endre" />
  </form>
</div>
`;
